src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java

```diff
@@ -112,42 +112,41 @@ protected void validateMovement(Movement movement, boolean checkReference) throw
 		}
 
 		// Check Lot
-		if (!isAutomaticLotOut()) {
-			Lot lot = movement.getLot();
-			if (lot != null) {
+		Lot lot = movement.getLot();
+		if (lot != null) {
 
-				if (lot.getCode().length() >= 50) {
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thelotidistoolongmax50chars.msg")));
-				}
+			if (lot.getCode().length() >= 50) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thelotidistoolongmax50chars.msg")));
+			}
 
-				if (lot.getDueDate() == null) {
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.insertavalidduedate.msg")));
-				}
+			if (lot.getDueDate() == null) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.insertavalidduedate.msg")));
+			}
 
-				if (lot.getPreparationDate() == null) {
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.insertavalidpreparationdate.msg")));
-				}
+			if (lot.getPreparationDate() == null) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.insertavalidpreparationdate.msg")));
+			}
 
-				if (lot.getPreparationDate() != null && lot.getDueDate() != null && lot.getPreparationDate().isAfter(lot.getDueDate())) {
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thepreparationdatecannotbyaftertheduedate.msg")));
-				}
+			if (lot.getPreparationDate() != null && lot.getDueDate() != null && lot.getPreparationDate().isAfter(lot.getDueDate())) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.thepreparationdatecannotbyaftertheduedate.msg")));
 			}
+		}
+		List<Integer> medicalIds = ioOperations.getMedicalsFromLot(lot.getCode());
```

Comment:
Shouldn't this be inside the `if (lot != null)` check?  If `lot` is `null` then `log.getCode()` will cause a NPE.

Reply:
Good catch, all of them indeed... let me revise the logic. Thanks!

Reply:
@dbmalkovsky please check now. Added also https://github.com/informatici/openhospital-gui/pull/1936. Thanks!